lib/miq_ldap.rb

```diff
@@ -303,7 +308,9 @@ def fqusername(username)
 
   def get_user_object(username, user_type = nil)
     user_type ||= @user_type.split("-").first
-    user_type = "dn" if self.is_dn?(username)
+    user_type = "dn" if self.dn?(username)
+    user_type = "upn" if self.upn?(username)
```

Comment:
this needs changing, I believe rdn's in LDAP can have the @ character.  so if you have cn=joev@redhat,cn=users,....  the logic in 312 will return true but would actually be false. You could use an if/else if structure.

Reply:
@abellotti Thank you. Will do!